crates/bevy_transform/src/hierarchy/hierarchy_maintenance_system.rs

```diff
@@ -45,20 +47,28 @@ pub fn parent_update_system(
             commands.entity(entity).insert(PreviousParent(parent.0));
         };
 
-        // Add to the parent's `Children` (either the real component, or
-        // `children_additions`).
-        if let Ok(mut new_parent_children) = children_query.get_mut(parent.0) {
-            // This is the parent
-            // PERF: Ideally we shouldn't need to check for duplicates
-            if !(*new_parent_children).0.contains(&entity) {
-                (*new_parent_children).0.push(entity);
+        match children_query.get_mut(parent.0) {
+            Ok(mut new_parent_children) => {
+                // This is the parent
+                // PERF: Ideally we shouldn't need to check for duplicates
+                if !(*new_parent_children).0.contains(&entity) {
+                    (*new_parent_children).0.push(entity);
+                }
+            }
+            Err(bevy_ecs::query::QueryEntityError::QueryDoesNotMatch) => {
+                // The parent doesn't have a children component, lets add it
+                children_additions
+                    .entry(parent.0)
+                    .or_insert_with(Default::default)
+                    .push(entity);
+            }
+            Err(bevy_ecs::query::QueryEntityError::NoSuchEntity) => {
+                // Our parent does not exist, so we should no longer exist.
```

Comment:
Is this actually true? A parent no longer existing doesn't _necessarily_ imply that the descendants should be despawned? Maybe the intent was to move the children to the top of the hierarchy?

Reply:
Like I say in the PR body, we cannot reasonably check whether `despawn` or `despawn_recursive` was called on the parent entity.

In my mind, a parent with all its children is an atomic unit, so it's much more likely that the user wants to despawn this child

But in general, this messiness is a consequence of how messy our entire parent/child system is; that is, I doubt this will be the final state. However, this patch makes the current state nicer, fixing an issue which several users ran into.